.github/workflows/release-reusable-build.yaml

```
@@ -0,0 +1,194 @@
+          if [ "${{ inputs.is_anchor }}" = "true" ]; then
+            echo "build-type=anchor" >> $GITHUB_OUTPUT
+          else
+            echo "build-type=vanilla" >> $GITHUB_OUTPUT
+          fi
+          echo "ref-used=${{ inputs.ref }}" >> $GITHUB_OUTPUT
+
+      - name: Set artifact name
+        id: set-artifact-name
+        run: |
+          echo "name=solana-programs" >> $GITHUB_OUTPUT
+          echo "path=target/deploy" >> $GITHUB_OUTPUT
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.85.1
+
+      - name: Install Solana CLI
+        run: |
+          sh -c "$(curl -sSfL https://release.solana.com/stable/install)"
+          echo "$HOME/.local/share/solana/install/active_release/bin" >> $GITHUB_PATH
+
+      - name: Install solana-verify
+        run: cargo install solana-verify
+
+      - name: Cache Cargo registry
+        uses: actions/cache@v4
+        with:
+          path: ~/.cargo/registry
+          key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-registry-
+
+      - name: Cache Cargo index
+        uses: actions/cache@v4
+        with:
+          path: ~/.cargo/git
+          key: ${{ runner.os }}-cargo-git-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-git-
+
+      - name: Cache target directory
+        uses: actions/cache@v4
+        with:
+          path: target
+          key: ${{ runner.os }}-target-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-target-
+
+      - name: Build xtask
+        run: cargo build -p xtask
+
+      - name: Update program IDs
+        if: inputs.update-ids
+        run: cargo xtask update-ids
+
+      - name: Commit updated IDs
+        if: inputs.update-ids && inputs.commit-changes
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git add -A
+          if git diff --staged --quiet; then
+            echo "No changes to commit"
+          else
+            git commit -m "chore: update program IDs [skip ci]"
+            git push
+          fi
+
+      - name: Build Solana programs
+        run: |
+          # Extract package names from release-plz.toml and build each one
+          PROGRAMS=$(grep -A 4 '^\[\[package\]\]' release-plz.toml | grep '^name = ' | sed 's/name = "\(.*\)"/\1/')
+
+          echo "Building programs from release-plz.toml:"
+          echo "$PROGRAMS"
+          echo ""
+
+          for program in $PROGRAMS; do
+            # Convert kebab-case to snake_case for library name
+            library_name=$(echo "$program" | tr '-' '_')
+            echo "Building $program (library: $library_name)..."
+            solana-verify build --base-image $BASE_IMAGE --library-name $library_name
```

```suggestion

            solana-verify build --base-image $BASE_IMAGE --library-name $library_name

```
the main cmd is a bit more clear